implementations/rust/ockam/ockam_transport_udp/src/puncture/rendezvous_service/rendezvous.rs

```diff
@@ -0,0 +1,208 @@
+
+impl RendezvousServiceWorker {
+    fn new() -> Self {
+        Self {}
+    }
+
+    /// Extract from `return route` everything just before we received the
+    /// message via UDP
+    fn parse_route(return_route: &Route) -> Route {
+        let addrs = return_route
+            .iter()
+            .skip_while(|x| x.transport_type() != UDP)
+            .cloned();
+
+        let mut res = Route::new();
+        for a in addrs {
+            res = res.append(a);
+        }
+        res.into()
+    }
+
+    /// Extract first UDP Address from `return route`
+    fn get_udp_address(return_route: &Route) -> Option<Address> {
+        return_route
+            .iter()
+            .find(|&x| x.transport_type() == UDP)
+            .cloned()
+    }
+
+    /// Handle Update request
+    fn handle_get_my_address(&mut self, return_route: &Route) -> Option<String> {
+        Self::get_udp_address(return_route).map(|a| a.address().to_string())
+    }
+}
+
+#[async_trait]
+impl Worker for RendezvousServiceWorker {
+    type Message = RendezvousRequest;
+    type Context = Context;
+
+    async fn handle_message(
+        &mut self,
+        ctx: &mut Context,
+        msg: Routed<Self::Message>,
+    ) -> Result<()> {
+        debug!(
+            "Received message: {:?} from {}",
+            msg,
+            Self::parse_route(&msg.return_route())
+        );
+        let return_route = msg.return_route();
+        match msg.into_body()? {
+            RendezvousRequest::Ping => {
+                ctx.send(return_route, RendezvousResponse::Pong).await?;
+            }
+            RendezvousRequest::GetMyAddress => {
+                let res = self.handle_get_my_address(&return_route);
+                match res {
+                    Some(udp_address) => {
+                        info!("{} got its public address", udp_address);
+                        ctx.send(return_route, RendezvousResponse::GetMyAddress(udp_address))
+                            .await?;
+                    }
+                    None => {
+                        // This could happen if a client erroneously contacts this service over TCP not UDP, for example
+                        warn!(
+                            "Return route has no UDP part, will not return address map: {:?}",
+                            return_route
+                        );
+                        // Ignore issue. There's no (current) way to inform sender.
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::RendezvousServiceWorker;
+    use crate::puncture::rendezvous_service::{RendezvousRequest, RendezvousResponse};
+    use crate::{RendezvousService, UdpBind, UdpBindArguments, UdpBindOptions, UdpTransport, UDP};
+    use ockam_core::{route, Result, Route, TransportType};
+    use ockam_node::Context;
+
+    #[test]
+    fn parse_route() {
+        assert_eq!(
+            route![(UDP, "c")],
```

Comment:
That's a minor thing, but I think we should stick to `assert_eq!(actual, expected)` like [most of the rust docs](https://doc.rust-lang.org/std/vec/struct.Vec.html#examples-16), even if it's not officially mandated.